### PR TITLE
Possible fix for server crash

### LINKED ISSUE #1598: Crash Report:  crash.log.092720151915
Here is a new one (from earlier today).

```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Error: signal 6:
/usr/local/bin/servatrice.reg(_ZN13SignalHandler14sigSegvHandlerEi+0x1f)[0x5046ef]
/lib64/libpthread.so.0(+0xf130)[0x7f0765a86130]
/lib64/libc.so.6(gsignal+0x37)[0x7f076447d5d7]
/lib64/libc.so.6(abort+0x148)[0x7f076447ecc8]
/lib64/libstdc++.so.6(_ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x165)[0x7f0764d819b5]
/lib64/libstdc++.so.6(+0x5e926)[0x7f0764d7f926]
/lib64/libstdc++.so.6(+0x5e953)[0x7f0764d7f953]
/lib64/libstdc++.so.6(+0x5eb73)[0x7f0764d7fb73]
/lib64/libstdc++.so.6(_Znwm+0x7d)[0x7f0764d8010d]
/lib64/libstdc++.so.6(_ZNSs4_Rep9_S_createEmmRKSaIcE+0x59)[0x7f0764ddec29]
/lib64/libstdc++.so.6(_ZNSs4_Rep8_M_cloneERKSaIcEm+0x1b)[0x7f0764ddf86b]
/lib64/libstdc++.so.6(_ZNSsC1ERKSs+0x5c)[0x7f0764ddff3c]
/usr/local/bin/servatrice.reg(_ZN6google8protobuf8internal14ArenaStringPtr21CreateInstanceNoArenaEPKSs+0x27)[0x4df127]
/usr/local/bin/servatrice.reg(_ZN22ServerInfo_ChatMessage9MergeFromERKS_+0x7b)[0x605b6b]
/usr/local/bin/servatrice.reg(_ZN22Server_ProtocolHandler11cmdJoinRoomERK16Command_JoinRoomR17ResponseContainer+0x6e1)[0x556a81]
/usr/local/bin/servatrice.reg(_ZN22Server_ProtocolHandler30processSessionCommandContainerERK16CommandContainerR17ResponseContainer+0x177)[0x557207]
/usr/local/bin/servatrice.reg(_ZN22Server_ProtocolHandler23processCommandContainerERK16CommandContainer+0xa6)[0x5578d6]
/usr/local/bin/servatrice.reg(_ZN21ServerSocketInterface10readClientEv+0x18e)[0x4f989e]
/usr/local/bin/servatrice.reg[0x51e3f5]
/lib64/libQt5Core.so.5(_ZN11QMetaObject8activateEP7QObjectiiPPv+0x846)[0x7f07652e6166]
/lib64/libQt5Network.so.5(+0xd1328)[0x7f07668e1328]
/lib64/libQt5Network.so.5(+0xde9e1)[0x7f07668ee9e1]
/lib64/libQt5Core.so.5(_ZN16QCoreApplication6notifyEP7QObjectP6QEvent+0x5d)[0x7f07652b562d]
/lib64/libQt5Core.so.5(_ZN16QCoreApplication14notifyInternalEP7QObjectP6QEvent+0x85)[0x7f07652b52d5]
/lib64/libQt5Core.so.5(+0x2e8255)[0x7f0765310255]
/lib64/libglib-2.0.so.0(g_main_context_dispatch+0x15a)[0x7f0763f0e99a]
/lib64/libglib-2.0.so.0(+0x49ce8)[0x7f0763f0ece8]
/lib64/libglib-2.0.so.0(g_main_context_iteration+0x2c)[0x7f0763f0ed9c]
/lib64/libQt5Core.so.5(_ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x7b)[0x7f076530f2db]
/lib64/libQt5Core.so.5(_ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE+0x12b)[0x7f07652b313b]
/lib64/libQt5Core.so.5(_ZN7QThread4execEv+0xb8)[0x7f07650c7038]
/lib64/libQt5Core.so.5(+0xa3ddf)[0x7f07650cbddf]
/lib64/libpthread.so.0(+0x7df5)[0x7f0765a7edf5]
/lib64/libc.so.6(clone+0x6d)[0x7f076453e1ad]

```

### PR DESCRIPTION
**Pull request by ctrlaltca**
Adding a mutex around chatHistory should fix the race condition that leads to the crash explained in #1598

Fix #1598

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,6 +19,10 @@ if(POLICY CMP0048)
     cmake_policy(SET CMP0048 OLD)
 endif()
 
+if(POLICY CMP0064)
+    cmake_policy(SET CMP0064 OLD)
```
Comment (ctrlaltca) [inline CMakeLists.txt]: Unrelated: avoid a warning on cmake >= 3.4

**Comment by ctrlaltca** [comment]
I tested this locally, but it's hard for me to reproduce a thousand users talking and joining the room..

**Comment by Daenyth** [comment]
The python script for registering might be able to do it.

**Comment by ctrlaltca** [comment]
Writing a fuzzer is a nice idea

**Comment by woogerboy21** [comment]
FYI, Ive put this on the woogerworks testing server if anyone wants too connect.  Seems to function (as i kind of expected) but I didnt put any major load on the server.

**Comment by ctrlaltca** [comment]
Thank you

**Pull request merged**